spec/models/user_spec.rb

```diff
@@ -965,10 +965,14 @@
       end
 
       it 'destroys SuspendedEmail records for each email address' do
-        email_address = user.email_addresses.last
+        email_addresses = user.email_addresses
+        email_address = email_addresses.last
+        expect(email_addresses.count).to eq 1
         expect { user.reinstate! }.
-          to(change { SuspendedEmail.find_with_email(email_address.email) }.
+          to(change { SuspendedEmail.find_with_email(email_address.email)&.email_address }.
             from(email_address).to(nil))
+        expect(email_addresses.reload.last).to_not be_nil
+        expect(email_addresses.reload.count).to eq 1
```

Comment:
another check we could do is to make sure email addresses is still present
```suggestion
        expect(user.email_addresses.reload.last).to be_present
```